test/tests/jobs/RetryJobTest.cpp

```diff
@@ -334,6 +335,37 @@ struct RetryJobTest : tpunit::TestFixture {
         ASSERT_EQUAL(difftime(nextRunTime, createdTime), 3600);
     }
 
+    // Retry a job with a repeat that uses the custom "START OF HOUR" modifier
+    void hasRepeatStartOfHour() {
+        uint64_t now = STimeNow();
+
+        // Create the job
+        SData command("CreateJob");
+        command["name"] = "job";
+        command["repeat"] = "STARTED, START OF DAY, +5 MINUTES, START OF HOUR";
+        command["nextRun"] = now;
+        STable response = tester->executeWaitVerifyContentTable(command);
+        string jobID = response["jobID"];
+
+        // Get the job
+        command.clear();
+        command.methodLine = "GetJob";
+        command["name"] = "job";
+        tester->executeWaitVerifyContent(command);
+
+        // Retry it
+        command.clear();
+        command.methodLine = "RetryJob";
+        command["jobID"] = jobID;
+        tester->executeWaitVerifyContent(command);
+
+        // Confirm nextRun is at the beginning of the day and not 5 minutes after
+        SQResult result;
+        tester->readDB("SELECT nextRun FROM jobs WHERE jobID = " + jobID + ";", result);
+        ASSERT_EQUAL(result.size(), 1);
+        ASSERT_EQUAL(result[0][0], SComposeTime("%Y-%m-%d 00:00:00", now));
```

Comment:
Also can you add a test where start of hour is not the last part to confirm its working well. Like for,
```
"STARTED, START OF DAY, START OF HOUR, +5 MINUTES"
```
and confirm the data is 5 minutes after start of hour. Or if you prefer you can update this existing one so `START OF HOUR` is not the last part.

Reply:
Yep, I will add one.